### PR TITLE
When HHTP client gets an HHTP Error, log it!

### PR DESCRIPTION
**Pull request by DebugBill**
Very useful to diagnose authentication issues between devices, trust me :-)

### CODE REVIEW

File: httpclient/HTTPClient.cpp [hunk 1]
```diff
@@ -115,7 +116,34 @@ bool HTTPClient::GETBinary(const std::string &url, const std::vector<std::string
...
+                                case 404:
+                                        _log.Log(LOG_ERROR,"HTTP 404: Not Found");
+                                        break;
+                                case 500:
+                                        _log.Log(LOG_ERROR,"HTTP 500: Internal Server Error");
+                                        break;
+                                case 503:
+                                        _log.Log(LOG_ERROR,"HTTP 503: Service Unavailable");
+                                        break;
+                        }
```
Comment (gizmocuz) [inline httpclient/HTTPClient.cpp]: Thanks!
Could you also add a default case handler that displays the response code?

**Review by gizmocuz — COMMENTED** [review]
Thanks

File: httpclient/HTTPClient.cpp (hunk 1 above)
Comment (DebugBill) [reply httpclient/HTTPClient.cpp]: You're right, it added

**Comment by gizmocuz** [comment]
Thanks!

**Pull request merged**